### PR TITLE
Remove ICustomMarshaler from CoreFX so that it is forwarded from CoreLib

### PR DESCRIPTION
**Pull request by shrah**
Fix for https://github.com/dotnet/coreclr/issues/9287

@tijoytom @jkotas

### CODE REVIEW

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs [hunk 1]
```diff
@@ -1,129 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-
-namespace System.Runtime.InteropServices
-{
-    [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class | AttributeTargets.Interface, Inherited = false)]
-    public sealed class AutomationProxyAttribute : Attribute
```
Comment (jkotas) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: I think these attributes should stay here - they should be removed from CoreLib instead.

Comment (shrah) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Attributes.cs]: Done. I will send another PR in CoreCLR removing these attributes.

**Comment by eerhardt** [comment]
Should there be a test added for ICustomMarshaler?  Given the scenario was completely broken, it seems like we should add a test.

**Comment by tijoytom** [comment]
I think so , @shrah can you please add the repro as a test.

**Pull request merged**